Fix unreliable tests

Travis builds are failing intermittently. Some sample output of the failures is below.

```
ok  	github.com/prebid/prebid-server	0.078s
body {"account_id":"","tid":"","cache_markup":0,"sort_bids":0,"max_key_length":0,"secure":0,"timeout_millis":0,"ad_units":[{"code":"first-tag","is_top_frame":0,"sizes":[{"w":300,"h":250}],"bids":[{"bidder":"audienceNetwork","bid_id":"random-id-from-pbjs-0","params":{"placementId":"987654321_999998888"}}],"config_id":""},{"code":"second-tag","is_top_frame":0,"sizes":[{"w":300,"h":250}],"bids":[{"bidder":"audienceNetwork","bid_id":"random-id-from-pbjs-1","params":{"placementId":"987654321_66775544"}}],"config_id":""}],"is_debug":false,"app":null,"device":null,"user":null,"Start":"0001-01-01T00:00:00Z"}
--- FAIL: TestFacebookBasicResponse (0.01s)
	facebook_test.go:273: Should not have gotten an error: context deadline exceeded
&{ [{div-adunit-1 0xc4203c2800 <nil> <nil>  0 1001 0  0 [] <nil> []}] 0xc420331860 <nil> 0xc420080d00 0xc4202a6280 0 1 0 [] 0 [] [] [] 0xc4201dd180 <nil> []}
body {"account_id":"","tid":"","cache_markup":0,"sort_bids":0,"max_key_length":0,"secure":0,"timeout_millis":0,"ad_units":[{"code":"first-tag","is_top_frame":0,"sizes":[{"w":300,"h":600},{"w":300,"h":250}],"bids":[{"bidder":"rubicon","bid_id":"random-id-from-pbjs-0","params":{"zoneId":8394,"siteId":283282,"accountId":7891}}],"config_id":""},{"code":"second-tag","is_top_frame":0,"sizes":[{"w":300,"h":600},{"w":300,"h":250}],"bids":[{"bidder":"rubicon","bid_id":"random-id-from-pbjs-1","params":{"zoneId":8395,"siteId":283282,"accountId":7891}}],"config_id":""}],"is_debug":false,"app":null,"device":null,"user":null,"Start":"0001-01-01T00:00:00Z"}
Request {"id":"","imp":[{"id":"second-tag","banner":{"w":300,"h":600,"format":[{"w":300,"h":600},{"w":300,"h":250}],"ext":{"rp":{"size_id":10,"alt_size_ids":[15],"mime":"text/html"}}},"ext":{"rp":{"zone_id":8395}}}],"site":{"domain":"nytimes.com","page":"https://www.nytimes.com/2017/05/04/movies/guardians-of-the-galaxy-2-review-chris-pratt.html?hpw\u0026rref=movies\u0026action=click\u0026pgtype=Homepage\u0026module=well-region\u0026region=bottom-well\u0026WT.nav=bottom-well\u0026_r=0","publisher":{"ext":{"rp":{"account_id":7891}}},"ext":{"rp":{"site_id":283282}}},"device":{"ua":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/603.1.30 (KHTML, like Gecko) Version/10.1 Safari/603.1.30","ip":"25.91.96.36"},"user":{"buyeruid":"need-an-actual-fb-id"},"at":1,"source":{"fd":1}}
Request {"id":"","imp":[{"id":"first-tag","banner":{"w":300,"h":600,"format":[{"w":300,"h":600},{"w":300,"h":250}],"ext":{"rp":{"size_id":10,"alt_size_ids":[15],"mime":"text/html"}}},"ext":{"rp":{"zone_id":8394}}}],"site":{"domain":"nytimes.com","page":"https://www.nytimes.com/2017/05/04/movies/guardians-of-the-galaxy-2-review-chris-pratt.html?hpw\u0026rref=movies\u0026action=click\u0026pgtype=Homepage\u0026module=well-region\u0026region=bottom-well\u0026WT.nav=bottom-well\u0026_r=0","publisher":{"ext":{"rp":{"account_id":7891}}},"ext":{"rp":{"site_id":283282}}},"device":{"ua":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/603.1.30 (KHTML, like Gecko) Version/10.1 Safari/603.1.30","ip":"25.91.96.36"},"user":{"buyeruid":"need-an-actual-fb-id"},"at":1,"source":{"fd":1}}
Request {"id":"","imp":[{"id":"first-tag","banner":{"w":300,"h":600,"format":[{"w":300,"h":600},{"w":300,"h":250}],"ext":{"rp":{"size_id":10,"alt_size_ids":[15],"mime":"text/html"}}},"ext":{"rp":{"zone_id":8394}}}],"site":{"domain":"nytimes.com","page":"https://www.nytimes.com/2017/05/04/movies/guardians-of-the-galaxy-2-review-chris-pratt.html?hpw\u0026rref=movies\u0026action=click\u0026pgtype=Homepage\u0026module=well-region\u0026region=bottom-well\u0026WT.nav=bottom-well\u0026_r=0","publisher":{"ext":{"rp":{"account_id":7891}}},"ext":{"rp":{"site_id":283282}}},"device":{"ua":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/603.1.30 (KHTML, like Gecko) Version/10.1 Safari/603.1.30","ip":"25.91.96.36"},"user":{"buyeruid":"need-an-actual-fb-id"},"at":1,"source":{"fd":1}}
Request {"id":"","imp":[{"id":"second-tag","banner":{"w":300,"h":600,"format":[{"w":300,"h":600},{"w":300,"h":250}],"ext":{"rp":{"size_id":10,"alt_size_ids":[15],"mime":"text/html"}}},"ext":{"rp":{"zone_id":8395}}}],"site":{"domain":"nytimes.com","page":"https://www.nytimes.com/2017/05/04/movies/guardians-of-the-galaxy-2-review-chris-pratt.html?hpw\u0026rref=movies\u0026action=click\u0026pgtype=Homepage\u0026module=well-region\u0026region=bottom-well\u0026WT.nav=bottom-well\u0026_r=0","publisher":{"ext":{"rp":{"account_id":7891}}},"ext":{"rp":{"site_id":283282}}},"device":{"ua":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_4) AppleWebKit/603.1.30 (KHTML, like Gecko) Version/10.1 Safari/603.1.30","ip":"25.91.96.36"},"user":{"buyeruid":"need-an-actual-fb-id"},"at":1,"source":{"fd":1}}
Error in rubicon adapter: context deadline exceededError in rubicon adapter: context deadline exceededFAIL
```

Expected: Unit tests always pass or fail reliably.

This popped up again in [this build](https://travis-ci.org/prebid/prebid-server/jobs/282374329)

```
--- FAIL: TestFacebookBasicResponse (0.03s)
	facebook_test.go:275: Should not have gotten an error: context deadline exceeded
body {"account_id":"","tid":"","cache_markup":0,"sort_bids":0,"max_key_length":0,"secure":0,"timeout_millis":0,"ad_units":[{"code":"first-tag","is_top_frame":0,"sizes":[{"w":320,"h":480}],"bids":[{"bidder":"lifestreet","bid_id":"random-id-from-pbjs-0","params":{"slot_tag":"slot123.123"}}],"config_id":"","media_types":null,"instl":0},{"code":"second-tag","is_top_frame":0,"sizes":[{"w":320,"h":480}],"bids":[{"bidder":"lifestreet","bid_id":"random-id-from-pbjs-1","params":{"slot_tag":"slot122.122"}}],"config_id":"","media_types":null,"instl":0}],"is_debug":false,"app":{"bundle":"AppNexus.PrebidMobileDemo"},"device":{"ua":"Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_1 like Mac OS X) AppleWebKit/603.1.30 (KHTML, like Gecko) Mobile/14E8301","ip":"111.111.111.111","make":"Apple","model":"x86_64","connectiontype":1,"ifa":"6F3EA622-C2EE-4449-A97A-AE986D080C08"},"user":null,"sdk":null,"Start":"0001-01-01T00:00:00Z"}
```

--- a/exchange/bidder_test.go
+++ b/exchange/bidder_test.go
@@ -5,8 +5,8 @@ import (
 	"errors"
 	"net/http"
 	"net/http/httptest"
-	"sync/atomic"
 	"testing"
+	"time"
 
 	"github.com/mxmCherry/openrtb"
 	"github.com/prebid/prebid-server/adapters"
@@ -138,21 +138,15 @@ func TestMultiBidder(t *testing.T) {
 
 }
 
-// TestBidderTimeout makes sure that things work smoothly if the context is cancelled in the middle
-// of an auction.
-//
-// If the context is "live", bidderAdapter.doRequest() should send responses back successfully.
-// If the context dies mid-request, or has already died before the request started, it should return errors.
+// TestBidderTimeout makes sure that things work smoothly if the context expires before the Bidder
+// manages to complete its task.
 func TestBidderTimeout(t *testing.T) {
-	ctx, cancelFunc := context.WithCancel(context.Background())
-	counter := int32(0)
-	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		numCalls := atomic.AddInt32(&counter, 1)
-		if numCalls == 2 {
-			cancelFunc()
-			<-ctx.Done() // Fixes #369 (hopefully)
-		}
+	// Fixes #369 (hopefully): Define a context which has already expired
+	ctx, cancelFunc := context.WithDeadline(context.Background(), time.Now().Add(-7*time.Hour))
+	cancelFunc()
+	<-ctx.Done()
 
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(200)
 		if r.Method == "GET" {
 			w.Write([]byte("getBody"))
@@ -173,36 +167,11 @@ func TestBidderTimeout(t *testing.T) {
 		Method: "POST",
 		Uri:    server.URL,
 	})
-	if callInfo.err != nil {
-		t.Errorf("The first call should not result in an error.")
-	}
-	if callInfo.response.StatusCode != 200 {
-		t.Errorf("The first call should result in a 200. Got %d", callInfo.response.StatusCode)
-	}
-	if string(callInfo.response.Body) != "postBody" {
-		t.Errorf("The first call should return the body \"postBody\". Got %s", callInfo.response.Body)
-	}
-
-	callInfo = bidder.doRequest(ctx, &adapters.RequestData{
-		Method: "POST",
-		Uri:    server.URL,
-	})
 	if callInfo.err == nil {
-		t.Errorf("The bidder should report an error if the context is cancelled mid-request.")
+		t.Errorf("The bidder should report an error if the context has expired already.")
 	}
 	if callInfo.response != nil {
-		t.Errorf("There should be no response if the request was cancelled.")
-	}
-
-	callInfo = bidder.doRequest(ctx, &adapters.RequestData{
-		Method: "POST",
-		Uri:    server.URL,
-	})
-	if callInfo.err == nil {
-		t.Errorf("The bidder should report an error if the context was cancelled before the request was made.")
-	}
-	if callInfo.response != nil {
-		t.Errorf("There should be no response if the context had already been cancelled.")
+		t.Errorf("There should be no response if the request never completed.")
 	}
 }
 

